From 93f2d2d55a1ceb2f7a495e79eebd4a79290c57e0 Mon Sep 17 00:00:00 2001
From: "Jan Chren (rindeal)" <dev.rindeal@gmail.com>
Date: Sun, 7 Jan 2018 23:02:18 +0100
Subject: [PATCH] make Makefile more packaging friendly

---
 Makefile | 35 +++++++++++++++++++----------------
 1 file changed, 19 insertions(+), 16 deletions(-)

diff --git a/Makefile b/Makefile
index e8224d8..7210fbe 100644
--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,11 @@
 VERSION = 1.6
 
 PREFIX ?= /usr/local
-MANPREFIX = $(PREFIX)/share/man
+MANPREFIX ?= $(PREFIX)/share/man
+STRIP ?= strip
 
-CFLAGS += -O3 -Wall -Wextra -Wno-unused-parameter
+CFLAGS ?= -O3
+CFLAGS += -Wall -Wextra -Wno-unused-parameter
 LDLIBS = -lreadline
 
 ifeq ($(shell pkg-config ncursesw && echo 1),1)
@@ -23,30 +25,31 @@ all: $(BIN) $(PLAYER)
 $(SRC): nnn.h
 
 $(BIN): $(SRC)
-	$(CC) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(LDLIBS)
-	strip $@
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LDLIBS)
 
 debug: $(SRC)
-	$(CC) -DDEBUGMODE -g $(CFLAGS) -o nnn $^ $(LDFLAGS) $(LDLIBS)
+	$(CC) -DDEBUGMODE -g $(CFLAGS) $(LDFLAGS) -o $(BIN) $^ $(LDLIBS)
 
 install: all
-	mkdir -p $(DESTDIR)$(PREFIX)/bin
-	cp -f $(BIN) $(DESTDIR)$(PREFIX)/bin
-	cp -f $(PLAYER) $(DESTDIR)$(PREFIX)/bin
-	mkdir -p $(DESTDIR)$(MANPREFIX)/man1
-	cp -f $(BIN).1 $(DESTDIR)$(MANPREFIX)/man1
+	install -m 0755 -D -t $(DESTDIR)$(PREFIX)/bin $(BIN) $(PLAYER)
+	install -m 0644 -D -t $(DESTDIR)$(MANPREFIX)/man1 $(BIN).1
 
 uninstall:
-	rm -f $(DESTDIR)$(PREFIX)/bin/$(BIN)
-	rm -f $(DESTDIR)$(PREFIX)/bin/$(PLAYER)
-	rm -f $(DESTDIR)$(MANPREFIX)/man1/$(BIN).1
+	$(RM) $(DESTDIR)$(PREFIX)/bin/$(BIN)
+	$(RM) $(DESTDIR)$(PREFIX)/bin/$(PLAYER)
+	$(RM) $(DESTDIR)$(MANPREFIX)/man1/$(BIN).1
+
+strip: $(BIN)
+	$(STRIP) $^
 
 dist:
 	mkdir -p nnn-$(VERSION)
-	cp $(DISTFILES) nnn-$(VERSION)
+	$(CP) $(DISTFILES) nnn-$(VERSION)
 	tar -cf nnn-$(VERSION).tar nnn-$(VERSION)
 	gzip nnn-$(VERSION).tar
-	rm -rf nnn-$(VERSION)
+	$(RM) -r nnn-$(VERSION)
 
 clean:
-	rm -f $(BIN) nnn-$(VERSION).tar.gz
+	$(RM) -f $(BIN) nnn-$(VERSION).tar.gz
+
+.PHONY: install uninstall clean debug strip dist
