From 2fd58155464fb4e753e8789c923aa640c1336127 Mon Sep 17 00:00:00 2001
From: "Jan Chren (rindeal)" <dev.rindeal@gmail.com>
Date: Mon, 18 Jun 2018 15:51:27 +0200
Subject: [PATCH] fix unguarded ENABLE_SUBIDS code

---
 libmisc/prefix_flag.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libmisc/prefix_flag.c b/libmisc/prefix_flag.c
index 96b11faa..cd1eec47 100644
--- a/libmisc/prefix_flag.c
+++ b/libmisc/prefix_flag.c
@@ -129,7 +129,7 @@ extern const char* process_prefix_flag (const char* short_opt, int argc, char **
 		snprintf(spw_db_file, len, "%s/%s", prefix, SHADOW_FILE);
 		spw_setdbname(spw_db_file);
 		
-
+#ifdef ENABLE_SUBIDS
 		len = strlen(prefix) + strlen("/etc/subuid") + 2;
 		suid_db_file = xmalloc(len);
 		snprintf(suid_db_file, len, "%s/%s", prefix, "/etc/subuid");
@@ -139,6 +139,7 @@ extern const char* process_prefix_flag (const char* short_opt, int argc, char **
 		sgid_db_file = xmalloc(len);
 		snprintf(sgid_db_file, len, "%s/%s", prefix, "/etc/subgid");
 		sub_gid_setdbname(sgid_db_file);
+#endif
 
 		len = strlen(prefix) + strlen("/etc/login.defs") + 2;
 		def_conf_file = xmalloc(len);
