Subject: don't print message to sdtout if -q flag is used
From: Alex Mestiashvili <mailatgoogl@gmail.com>
Forwarded: https://sourceforge.net/p/hdparm/patches/48/
--- hdparm.orig/hdparm.c
+++ hdparm/hdparm.c
@@ -2258,7 +2258,8 @@
 	}
 	if (security_freeze) {
 		__u8 args[4] = {ATA_OP_SECURITY_FREEZE_LOCK,0,0,0};
-		printf(" issuing security freeze command\n");
+		if (!quiet)
+			printf(" issuing security freeze command\n");
 		if (do_drive_cmd(fd, args, 0)) {
 			err = errno;
 			perror(" HDIO_DRIVE_CMD(security_freeze) failed");
